src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/AbstractServerTask.groovy

```diff
@@ -211,6 +215,109 @@ abstract class AbstractServerTask extends AbstractTask {
         }
     }
 
+    protected void writePluginConfigXml(Project project) {
+        new File(project.buildDir, 'liberty-plugin-config.xml').withWriter { writer ->
+            def xml = new MarkupBuilder(writer)
+            xml.mkp.xmlDeclaration(version: "1.0", encoding: "UTF-8")
+            xml.'liberty-plugin-config' {
+                installDirectory (getInstallDir(project).toString())
+                serverDirectory (getServerDir(project).toString())
+                userDirectory (getUserDir(project).toString())
+
+                String serverOutputDir = getServerOutputDir(project)
+                if (serverOutputDir != null && !serverOutputDir.isEmpty()) {
+                    serverOutputDirectory (serverOutputDir)
+                } else {
+                    serverOutputDirectory (getServerDir(project).toString())
+                }
+
+                serverName (server.name)
+
+                if (server.configDirectory != null && server.configDirectory.exists()) {
+                    configDirectory (server.configDirectory.toString())
+                }
+
+                if (server.configFile != null && server.configFile.exists()) {
+                    configFile (server.configFile.toString())
+                }
+
+                if (server.bootstrapProperties != null && !server.bootstrapProperties.isEmpty()) {
+                    bootstrapProperties {
+                        server.bootstrapProperties.collect { k, v ->
+                            "$k" (v.toString())
+                        }
+                    }
+                } else if (server.bootstrapPropertiesFile != null && server.bootstrapPropertiesFile.exists()) {
+                    bootstrapPropertiesFile (server.bootstrapPropertiesFile.toString())
+                }
+
+                if (server.jvmOptions != null && !server.jvmOptions.isEmpty()) {
+                    jvmOptions {
+                        server.jvmOptions.collect { v ->
+                            "params" (v.toString())
+                        }
+                    }
+                } else if (server.jvmOptionsFile != null && server.jvmOptionsFile.exists()) {
+                    jvmOptionsFile (server.jvmOptionsFile.toString())
+                }
+
+                if (server.serverEnv != null && server.serverEnv.exists()) {
+                    serverEnv (server.serverEnv.toString())
+                }
+
+                appsDirectory (server.appsDirectory)
+                looseApplication (server.looseApplication)
+                stripVersion (server.stripVersion)
```

Comment:
All the server configuration is prime for refactoring to a method that takes a server object and the xml object you need to continue writing to the file. Not sure how that works with this groovy xml writer.